### ISSUE TITLE
Problem with removing branch from branches list

### ISSUE THREAD

**Issue by michalkortas** [desktop/desktop#8831]
Hi All, 
I found a small bug (or only inconvenience). When I try delete local branch with remote branch and this branch was deleted a moment earlier via github.com website, this branch is still on the branches list (GH Desktop) and remove action does not return any error / info message. 

This branch disappears from list only after fetching origin.

### Version & OS

GitHub Desktop 2.2.4
Windows 10 Pro

### Steps to reproduce the behavior

1. Go to github.com and delete branch
2. Go to GitHub Desktop and delete remote branch **without** fetching origin
3. Removing modal box disappears without any info / error message
4. Removed branch is still on the branches list

### Expected behavior

Information with removing error - "Branch does not exist on remote". Branch should be remove from branches list.

### Actual behavior

Removing modal box disappears without any info / error message. Removed branch is still on the branches list.

**Comment by niik** [comment]
Hey @michalkortas, thanks for opening this. This behavior was partly introduced in https://github.com/desktop/desktop/pull/1834 with the goal of preventing an error message or warning when attempting to delete a branch that no longer exists on the remote. At its core I think that's the right way to go here, i.e. we shouldn't show an error message. I do however agree with you that if our attempt to delete the remote branch fails because it doesn't exist on the remote then we should remove it locally as well (one could even argue that this is something that should be taken care of in Git itself).

I'm going to tentatively classify this as a lower priority bug since we haven't heard a lot of feedback about this behavior being confusing (I suspect many users don't do explicit branch cleanup locally). That said I still think it has the potential of being quite confusing so I'll take a quick look and see if this is some quick fix here that I could squeeze in in-between the holidays.

**Comment by michalkortas** [comment]
Hi @niik, thank you for your very fast reaction.

**Comment by H12M54AM** [comment]
Hi there. I have reached the same problem as michalkortas where I go that error on Github Desktop. I'm not sure if there is a solution already since not many people had the same problem, however it does suck not having Github delete files or rename files/folders locally on your machine, the same name or way on Github so everything is synced. Is there a particular place I can go to find the solution or A solution?
